ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -493,7 +499,9 @@ private void cleanupExpiredInFlightRequests() {
         Map.Entry<Integer, GetRequestInfo> entry = inFlightRequestsIterator.next();
         if (time.milliseconds() - entry.getValue().startTimeMs > routerConfig.routerRequestTimeoutMs) {
           onErrorResponse(entry.getValue().replicaId);
-          chunkException = new RouterException("Timed out waiting for responses", RouterErrorCode.OperationTimedOut);
+          responseHandler.onRequestResponseException(entry.getValue().replicaId,
+              new IOException("Timed out waiting for a response"));
+          chunkException = new RouterException("Timed out waiting for a response", RouterErrorCode.OperationTimedOut);
```

Comment:
same as above. The `IOException` does not contain any useful information in order to be embedded into the `RouterException`.